sdk/resourcemanager/azure-resourcemanager-redis/src/main/java/com/azure/resourcemanager/redis/models/RedisCache.java

```diff
@@ -581,6 +585,23 @@ enum RedisVersion {
         public String getValue() {
             return value;
         }
+
+        /**
+         * Gets the major version from the given redis version.
+         * @param redisVersion valid redis version, e.g. 6.0.14
+         * @return the major redis version, or null if the given redisVersion is invalid
+         */
+        public static RedisVersion fromVersion(String redisVersion) {
+            if (CoreUtils.isNullOrEmpty(redisVersion)) {
+                return null;
+            }
+            for (RedisVersion version : values()) {
+                if (redisVersion.startsWith(version.getValue())) {
```

Comment:
nit on startWith... when redis get to 41.0.0

Reply:
Right... Thanks...